### PR TITLE
Abstract class enforced

### PR DESCRIPTION
**Pull request by DEKHTIARJonathan**
### Motivation and Context
With the PR #519 I noticed that the class LayersConfig was used and thought as an abstract class containing various data.

This PR enforce this behavior to ensure that only one class named like this exist in the library. I also have implemented a unittest for it.

### CODE REVIEW

**Comment by zsdonghao** [comment]
hi, where is the `abc` from? it is a new dependence? we may need to add it into `requirement.txt`

**Comment by DEKHTIARJonathan** [comment]
Damned this doesn't work with Python 2.7...
I need to fix this ;)

File: tests/test_layers_core.py [hunk 1]
```diff
@@ -6,6 +6,16 @@
 import tensorlayer as tl
 
 
+class Core_Helpers_Test(unittest.TestCase):
+
+    def test_LayersConfig(self):
+        with self.assertRaises(TypeError):
+            tl.layers.LayersConfig()
```
[the remark this answers is not in the document]
Comment (DEKHTIARJonathan) [reply tests/test_layers_core.py]: This Issue is not an issue ... It's a unittest !

**Comment by DEKHTIARJonathan** [comment]
This PR can be merged ;)
@zsdonghao

**Comment by DEKHTIARJonathan** [comment]
You don't need to install abc @zsdonghao, it's a core module of python

**Pull request merged**